pyro/infer/tracegraph_elbo.py

```diff
@@ -92,20 +137,30 @@ def _compute_downstream_costs(model_trace, guide_trace,  #
 
 
 def _compute_elbo_reparam(model_trace, guide_trace):
+
+    # Compute a surrogate ELBO. In [1], this is simply the ELBO because
+    # (non)reparameterization is captured in the structure of the computation
+    # graph, and the required behavior under parameter differentiation emerges
+    # from that structure. Here the behavior of E_p[log(p)] under parameter
+    # differentiation is instead represented in site["score_parts"].
+
     elbo = 0.0
     surrogate_elbo = 0.0
 
-    # deal with log p(z|...) terms
+    # Bring log p(x, z|...) terms into both the ELBO and the surrogate
     for name, site in model_trace.nodes.items():
         if site["type"] == "sample":
             elbo += site["log_prob_sum"]
             surrogate_elbo += site["log_prob_sum"]
 
-    # deal with log q(z|...) terms
+    # Bring log q(z|...) terms into the ELBO, and effective terms into the
```

Comment:
what's an "effective term"?

Reply:
If reparameterizing a stochastic node inserted the transformation into the computation graph, say `z=t(epsilon, theta)`, the resulting cost terms `log q(epsilon|...)` naturally wouldn't contribute to the gradients. Here reparamerization is not explicit in the graph, instead by default it seems to be inferred from the attribute `has_rsample`. Then given that (non)reparameterization, the behavior under differentiation is stored in `score_parts`.

Coming from the paper, `site['score_parts'].entropy_term` felt like an effective term for `log q` given that the parameterization is implicit. 

Would it be useful to make these statements more precise?